File: src/C3.Catalogue/Brands/IBrandRepository.vb
Namespace Brands

    Public Interface IBrandRepository

        Function GetAll() As IList(Of Brand)
        Function FindByCode(code As String) As Brand
        Function IsCodeInUse(code As String) As Boolean
        Function IsReferencedByModel(code As String) As Boolean
        Sub Add(value As Brand)
        Sub Update(value As Brand)
        Sub Delete(code As String)

    End Interface

End Namespace

File: src/C3.Catalogue/Brands/IBrandRepository.cs
using System.Collections.Generic;

namespace C3.Catalogue.Brands
{
    public interface IBrandRepository
    {
        IList<Brand> GetAll();

        Brand FindByCode(string code);

        bool IsCodeInUse(string code);

        bool IsReferencedByModel(string code);

        void Add(Brand value);

        void Update(Brand value);

        void Delete(string code);
    }
}
